examples/rl_gsm8k/orchestrate_rl.py

```diff
@@ -126,15 +129,27 @@ def extract_tape_training_samples(
         sub_llm_calls = sorted(sub_llm_calls, key=lambda call: prompt_ids.index(call.prompt.id))
         for i, llm_call in enumerate(sub_llm_calls[::-1]):
             trace = agent.llm.make_training_text(llm_call.prompt, llm_call.output)
-            # Check if we will need the KL
-            if hasattr(cfg.finetune, "rl") and (
-                cfg.finetune.rl.kl_coef > 0 or cfg.finetune.rl.reward_minus_kl_coef > 0
-            ):
-                trace.logprobs = agent.llm.get_log_probs(trace.prompt_text, trace.output_text)  # type: ignore
+
+
+            hf_tokens = get_tokens_from_hf_tokenizer(agent.llm.tokenizer, llm_call.prompt, llm_call.output)
+
+            logprobs = []
+            vllm_tokens = []
+            if hasattr(llm_call.output, "logprobs"):
+                logprobs = llm_call.output.logprobs.content
+                logprobs = [c.logprob for c in logprobs]
+                vllm_tokens = [c.token for c in llm_call.output.logprobs.content]
+
+            if (strict and vllm_tokens != hf_tokens) or (not strict and len(logprobs) != llm_call.output_length_tokens):
```

Comment:
let's always be strict, should not be that much of a performance burden I suppose